data/Templates/eCR/DataType/_Identifier.liquid

```diff
@@ -1,13 +1,33 @@
-{% if Identifier.nullFlavor == null -%}
-    {% if Identifier.extension -%}
-        "system":{% include 'ValueSet/SystemReference' code: Identifier.root -%},
-        "value":"{{Identifier.extension}}",
-        "assigner":
-        {
-            "display":"{{Identifier.assigningAuthorityName}}"
+{
+{%- if Identifier.nullFlavor == null -%}
+    {%- assign oid_pattern = "^([0-2])((\.0)|(\.[1-9][0-9]*))*$" -%}
+    {%- assign uuid_pattern = "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$" -%}
+    {%- assign uid_pattern = uuid_pattern | append: "|" | append: oid_pattern -%}
+    {%- if Identifier.extension -%}
+        {%- assign oid_pattern_match = Identifier.root | match: oid_pattern | size -%}
+        {%- if oid_pattern_match == 1 -%}
+            {%- capture system -%}
+                {%- include 'ValueSet/SystemReference' code: Identifier.root -%}
+            {%- endcapture %}
+            "system": {{- system -}},
+            "value": "{{- Identifier.extension | remove_prefix: system -}}",
+        {%- else -%}
+            "system": "{{- Identifier.root | prepend_id -}}",
+            "value": "{{- Identifier.extension -}}",
```

Comment:
should this also remove prefix? (just in cases the root was already http-ified for whatever weird reason)